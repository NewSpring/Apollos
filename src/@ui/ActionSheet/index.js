export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
