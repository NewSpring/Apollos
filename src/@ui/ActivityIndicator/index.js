import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
