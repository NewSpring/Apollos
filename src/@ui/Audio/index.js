import React, { Component } from 'react';
import { Audio as ExpoAudio } from 'expo';
import PropTypes from 'prop-types';
import { Animated, View } from 'react-native';
import AudioPlay from './AudioPlay';
import AudioPause from './AudioPause';
import AudioSeeker from './AudioSeeker';

export default class Audio extends Component {
  static Play = AudioPlay;
  static Pause = AudioPause;
  static Seeker = AudioSeeker;

  static propTypes = {
    source: PropTypes.string,
    onReady: PropTypes.func,
    onError: PropTypes.func,
    onPlaybackReachedEnd: PropTypes.func,
    onPlay: PropTypes.func,
    onPause: PropTypes.func,
    onStop: PropTypes.func,
    onSeek: PropTypes.func,
    onSeeking: PropTypes.func,
    children: PropTypes.node,
    isPlaying: PropTypes.bool,
    style: PropTypes.any, // eslint-disable-line
  };

  static defaultProps = {
    onReady() {},
    onError(err) {
      // eslint-disable-next-line no-console
      console.error(err.message);
    },
    onPlaybackReachedEnd() {},
    onPlay() {},
    onPause() {},
    onStop() {},
    onSeek() {},
    onSeeking() {},
    children: null,
    isPlaying: false,
  };

  static childContextTypes = {
    play: PropTypes.func,
    stop: PropTypes.func,
    pause: PropTypes.func,
    seek: PropTypes.func,
    progress: PropTypes.object,
    positionMillis: PropTypes.object,
    seekingHandler: PropTypes.func,
  };

  getChildContext = () => ({
    play: this.play,
    stop: this.stop,
    pause: this.pause,
    seek: this.seek,
    progress: this.progressDriver,
    positionMillis: this.positionMillisDriver,
    seekingHandler: this.handleSeeking,
  });

  componentWillMount() {
    ExpoAudio.setAudioModeAsync({
      allowsRecordingIOS: false,
      interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
      playsInSilentModeIOS: true,
      shouldDuckAndroid: true,
      interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX,
    });

    this.loadSource();
  }

  componentDidUpdate({ source, isPlaying }) {
    if (this.props.source !== source) {
      this.loadSource();
    } else if (this.props.isPlaying !== isPlaying) {
      if (this.props.isPlaying) this.play();
      if (!this.props.isPlaying) this.pause();
    }
  }

  componentWillUnmount() {
    if (this.sound) this.sound.unloadAsync();
  }

  onPlaybackStatusUpdate = (soundStatus) => {
    this.positionMillisDriver.setValue(soundStatus.positionMillis);
    this.progressDriver.setValue((soundStatus.positionMillis / this.duration) || 0);

    if (soundStatus.didJustFinish) {
      this.props.onPlaybackReachedEnd(this.sound);
    }
  }

  positionMillisDriver = new Animated.Value(0);
  progressDriver = new Animated.Value(0);
  duration = 0;
  positionListener = undefined;
  previousSoundStatus = undefined;
  isReady = false;
  playState = this.props.isPlaying;

  play = async () => {
    if (this.sound) {
      try {
        await this.sound.playAsync();
        this.props.onPlay();
      } catch (err) {
        this.props.onError(err);
      }
    }
  }

  pause = async () => {
    if (this.sound) {
      try {
        await this.sound.pauseAsync();
        this.props.onPause();
      } catch (err) {
        this.props.onError(err);
      }
    }
  }

  stop = async () => {
    try {
      this.sound.stopAsync();
      this.props.onStop();
    } catch (err) {
      this.props.onError(err);
    }
  }

  seek = (percentageOfSong) => {
    const positionInMillis = this.duration * percentageOfSong;
    this.sound.setPositionAsync(positionInMillis);
    this.props.onSeek(positionInMillis);
  }

  handleSeeking = (percentageOfSong) => {
    const positionInMillis = this.duration * percentageOfSong;
    this.props.onSeeking(positionInMillis);
  }

  loadSource = async () => {
    const {
      source,
      onReady,
      onError,
    } = this.props;

    if (!source) return;
    const uri = source.replace(/^http:\/\/|^\/\//i, 'https://');

    try {
      if (this.sound && this.soundLoaded === this.sound) {
        await this.sound.unloadAsync();
      }

      const sound = this.sound = await new ExpoAudio.Sound(); // eslint-disable-line
      const soundStatus = await this.sound.loadAsync({ uri });
      this.soundLoaded = this.sound;

      if (sound === this.sound) { // since the above calls are promise base, there's a chance that
      // the current sound changes before this sound is loaded (pressing skip button quickly).
      // We don't want to start playing this sound in that scenario :)
        this.duration = soundStatus.durationMillis;
        this.sound.setOnPlaybackStatusUpdate(this.onPlaybackStatusUpdate);
        if (this.props.isPlaying) await this.play();
      }

      onReady();
    } catch (err) {
      onError(err);
    }
  }

  render() {
    return <View style={this.props.style}>{this.props.children}</View>;
  }
}
