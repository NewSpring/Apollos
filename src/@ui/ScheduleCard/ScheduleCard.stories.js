import React from 'react';

import { storiesOf } from '@storybook/react-native';
import FlexedView from '@ui/FlexedView';

import ScheduleCard from './';

storiesOf('ScheduleCard', module)
  .add('default', () => (
    <FlexedView>
      <ScheduleCard
        accountName="General Fund"
        amount={2.22}
        frequency="One Time"
        startDate="2018-01-18T00:00:00.000Z"
      />
    </FlexedView>
  ));
