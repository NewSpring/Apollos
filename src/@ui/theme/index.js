export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
