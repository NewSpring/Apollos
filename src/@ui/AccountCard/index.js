import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
