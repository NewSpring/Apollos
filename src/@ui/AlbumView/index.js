import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { View, Platform, StyleSheet } from 'react-native';
import PaddedView from '@ui/PaddedView';
import ProgressiveImage from '@ui/ProgressiveImage';
import { H5, H7 } from '@ui/typography';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

const Container = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
}), 'AlbumView.Container')(View);

const InfoWrapper = styled({
  flexDirection: 'row', alignItems: 'stretch',
}, 'AlbumView.InfoWrapper')(PaddedView);

const AlbumArt = styled({
  position: 'relative',
  width: '33%',
  aspectRatio: 1,
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '100%',
    },
  }),
}, 'AlbumView.AlbumArt')(ProgressiveImage);

const BlurredImage = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  opacity: theme.alpha.low,
}), 'AlbumView.BlurredImage')(ProgressiveImage);

const TitleWrapper = styled({
  width: '66%',
  justifyContent: 'space-around',
}, 'AlbumView.TitleWrapper')(PaddedView);

const Title = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Title')(H5);

const Artist = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Artist')(H7);

const enhance = compose(
  withThemeMixin({ type: 'dark' }),
  pure,
  setPropTypes({
    title: PropTypes.string,
    artist: PropTypes.string,
    blurredImage: ProgressiveImage.propTypes.source,
    albumImage: ProgressiveImage.propTypes.source,
    isLoading: PropTypes.bool,
  }),
);

const AlbumView = enhance(({
  title, artist, blurredImage, albumImage, isLoading = false,
}) => (
  <Container>
    {(blurredImage) ? <BlurredImage source={blurredImage} /> : null}
    <InfoWrapper>
      <AlbumArt isLoading={isLoading} source={albumImage} />
      <TitleWrapper>
        <Title>{title}</Title>
        <Artist>{artist}</Artist>
      </TitleWrapper>
    </InfoWrapper>
  </Container>
));

export default AlbumView;
