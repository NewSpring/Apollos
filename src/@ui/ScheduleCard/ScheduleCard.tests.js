import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '@ui/theme';
import ScheduleCard from './';

describe('the ScheduleCard component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <ScheduleCard
          accountName="accountName"
          amount={20}
          frequency="often"
          startDate="02/02/2020"
          iconSize={20}
          dateFormat="lll"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});

