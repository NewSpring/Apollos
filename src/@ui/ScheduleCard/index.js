import React from 'react';
import {
  View,
  TouchableWithoutFeedback,
} from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import moment from 'moment';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card from '@ui/Card';
import PaddedView from '@ui/PaddedView';
import { H5, H6, H3, H1, UIText } from '@ui/typography';
import Icon from '@ui/Icon';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    isLoading: PropTypes.bool,
    accountName: PropTypes.string,
    amount: PropTypes.string,
    frequency: PropTypes.string,
    startDate: PropTypes.string,
    iconSize: PropTypes.number,
    dateFormat: PropTypes.string,
  }),
  defaultProps({
    isLoading: false,
    accountName: null,
    amount: 0,
    frequency: null,
    startDate: null,
    iconSize: null,
    dateFormat: 'MMM D YYYY',
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const StyledH6 = styled(({ theme }) => ({
  color: theme.colors.text.link,
}))(H6);

const StyledH1 = styled(({ theme }) => ({
  color: theme.colors.text.primary,
}))(H1);

const ItalicText = styled(({ theme }) => ({
  color: theme.colors.text.secondary,
  fontStyle: 'italic',
}))(UIText);

const Row = styled({
  flexDirection: 'row',
  alignItems: 'center',
})(View);

const Spacer = styled(({ theme }) => ({
  height: theme.sizing.baseUnit,
}))(View);

const ScheduleCard = enhance(({
  isLoading,
  accountName,
  amount,
  frequency,
  startDate,
  iconSize,
  dateFormat,
  onPress,
  ...otherProps
}) => {
  const amountParts = amount.toFixed(2).split('.');
  return (
    <Card isLoading={isLoading} {...otherProps}>
      <PaddedView>
        <Row>
          <H3>{'$'}</H3>
          <StyledH1>{amountParts[0]}</StyledH1>
          <H3>{`.${amountParts[1]}`}</H3>
        </Row>
        <H5>{frequency}</H5>
        <H5>{accountName}</H5>
        <Spacer />
        <Row>
          <UIText>{'Start Date: '}</UIText>
          <ItalicText>{moment(startDate).format(dateFormat)}</ItalicText>
        </Row>
        <Spacer />
        <TouchableWithoutFeedback
          onPress={onPress}
        >
          <Row>
            <StyledH6>{'View Schedule Details'}</StyledH6>
            <Icon name="ArrowNext" size={iconSize} />
          </Row>
        </TouchableWithoutFeedback>
      </PaddedView>
    </Card>
  );
});

export default ScheduleCard;
