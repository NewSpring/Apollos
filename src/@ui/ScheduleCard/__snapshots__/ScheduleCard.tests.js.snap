// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ScheduleCard component should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "marginHorizontal": 10,
      "marginVertical": 5,
      "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Bold",
                "fontSize": 32.4,
                "lineHeight": 41.04,
                "paddingBottom": 2.7,
                "paddingTop": 5.4,
              }
            }
          >
            $
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Bold",
                "fontSize": 52.2,
                "lineHeight": 54.81,
                "paddingTop": 11.7,
              }
            }
          >
            20
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Bold",
                "fontSize": 32.4,
                "lineHeight": 41.04,
                "paddingBottom": 2.7,
                "paddingTop": 5.4,
              }
            }
          >
            .00
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#303030",
              "fontFamily": "Colfax-Bold",
              "fontSize": 20.02,
              "lineHeight": 25.33,
              "paddingBottom": 3.15,
              "paddingTop": 4.5,
            }
          }
        >
          often
        </Text>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Colfax-Regular",
                "fontSize": 14,
                "lineHeight": 15.78,
                "paddingBottom": 1.8,
                "paddingTop": 3.6,
              }
            }
          >
            Start Date:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            bold={false}
            ellipsizeMode="tail"
            italic={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "DroidSerif-RegularItalic",
                "fontSize": 20.02,
                "lineHeight": 33.11,
                "paddingBottom": 2.25,
                "paddingLeft": 10,
                "paddingTop": 1.8,
              }
            }
          >
            Jan 17, 2018 12:00 AM
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 10,
              "width": null,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#6bac43",
                "fontFamily": "Colfax-Bold",
                "fontSize": 15.75,
                "lineHeight": 17.75,
                "paddingBottom": 1.8,
                "paddingTop": 4.5,
              }
            }
          >
            View Schedule Details
          </Text>
          <Svg
            height={20}
            viewBox="0 0 24 24"
            width={20}
          >
            <Path
              d="M9.65 21L8 20.1l7.72-8.1L8 3.9 9.65 3l8.16 8.54c.27.28.27.64 0 .92L9.67 21z"
              fill="#6bac43"
            />
          </Svg>
        </View>
      </View>
    </View>
  </View>
</View>
`;
